src/components/TimeBasedLineChart.js

```diff
@@ -0,0 +1,753 @@
+  }
+
+  _getSplitData () {
+    const future = [];
+    const past = [];
+
+    this.props.data.forEach(item => {
+      const currentDate = moment.unix(item.timeStamp).startOf(this.props.rangeType);
+
+      if (currentDate.isBefore(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
+        past.push(item);
+      }
+
+      if (currentDate.isAfter(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
+        future.push(item);
+      }
+
+      if (currentDate.isSame(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
+        past.push(item);
+        future.push(item);
+      }
+    });
+
+    return [past, future];
+  }
+
+  _getYAxisTicks (data, numTicks = 4) {
+    const maxValue = d3.max(this.props.data, d => {
+      const value = d.above ? d.value + d.above : d.value + 1000;
+      const multiplier = value < 0 ? -1000 : 1000;
+
+      return Math.ceil(value / multiplier) * multiplier;
+    });
+
+    let minValue = d3.min(this.props.data, d => {
+      const value = d.below ? d.value + d.below : d.value - 1000;
+      const multiplier = value < 0 ? -1000 : 1000;
+
+      return Math.ceil(value / multiplier) * multiplier;
+    });
+
+    minValue = minValue > 0 ? 0 : minValue;
+
+    const ticks = [maxValue, minValue];
+    const interval = (maxValue - minValue) / numTicks;
+
+    for (let current = minValue; current < maxValue; current += interval) {
+      ticks.push(current);
+    }
+
+    if (this.props.alwaysShowZeroYTick) {
+      if (ticks.indexOf(0) === -1) {
+        ticks.push(0);
+      }
+    }
+
+    return ticks;
+  }
+
+  _getXScaleFunction () {
+    let maxDate = this.props.data[this.props.data.length - 1].timeStamp;
+    let minDate = this.props.data[0].timeStamp;
+
+    maxDate = moment.unix(maxDate).endOf(this.props.rangeType).unix();
+    minDate = moment.unix(minDate).startOf(this.props.rangeType).unix();
+
+    return d3.time.scale()
+      .range([0, this.state.adjustedWidth])
+      .domain([minDate, maxDate]);
+  }
+
+  _getXScaleValue (value) {
+    const xScale = this._getXScaleFunction();
+
+    return xScale(value);
+  }
+
+  _getYScaleFunction () {
+    const maxValue = d3.max(this.props.data, d => {
+      const value = d.above ? d.value + d.above : d.value + 1000;
+      const multiplier = value < 0 ? -1000 : 1000;
+
+      return Math.ceil(value / multiplier) * multiplier;
+    });
+
+    let minValue = d3.min(this.props.data, d => {
+      const value = d.below ? d.value + d.below : d.value - 1000;
+      const multiplier = value < 0 ? -1000 : 1000;
+
+      return Math.ceil(value / multiplier) * multiplier;
+    });
+
+    minValue = minValue > 0 ? 0 : minValue;
+
+    return d3.scale.linear()
+      .range([this.state.adjustedHeight, 0])
+      .domain([minValue, maxValue]);
+  }
+
+  _getYScaleValue (value) {
+    const yScale = this._getYScaleFunction();
+
+    return yScale(value) - 10;
```

Comment:
Accounts for an addition of 10 pixels to the bottom margin when setting up the base chart